scala/scala.bzl

```diff
@@ -494,6 +511,55 @@ def _scala_test_impl(ctx):
     _write_test_launcher(ctx, rjars)
     return _scala_binary_common(ctx, cjars, rjars)
 
+def _prep_grep_pattern_from(patterns):
```

Comment:
I added in the general comment about how the design works. Do you want another comment here?
This is another indication that maybe I should move the grepping to scala code